### PR TITLE
Add privileged flag to helm charts

### LINKED ISSUE #34211: Istio 1.11.0-beta.1 fails on OCP 4.7 - istio-cni-node pods crashing with "Error: failed to create UDS listener: listen unix /var/run/istio-cni/log.sock: bind: permission denied"
**Bug description**
When installing Istio 1.11.0-beta.1 on OCP 4.7, all istio-cni-node pods are crashing with "Error: failed to create UDS listener: listen unix /var/run/istio-cni/log.sock: bind: permission denied"

Full log attached.

[ ] Docs
[X ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Extensions and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure
[ ] Upgrade

**Expected behavior**
Istio is installed correctly

**Steps to reproduce the bug**

1. `oc adm policy add-scc-to-group anyuid system:serviceaccounts:istio-system`
2. run `istioctl manifest install ` command specified bellow

**Version** (include the output of `istioctl version --remote` and `kubectl version --short` and `helm version --short` if you used Helm)
OCP 4.7.19
Istio 1.11.0-beta.1

**How was Istio installed?**
`istioctl manifest install --skip-confirmation=true --set profile=demo  --set hub=gcr.io/istio-release  --set values.meshConfig.enableAutoMtls=true  --set values.gateways.istio-egressgateway.enabled=true --set values.gateways.istio-ingressgateway.enabled=true --set values.global.meshID=mesh-default --set values.global.multiCluster.clusterName=cluster-default --set values.global.network=network-default --set components.cni.enabled=true --set components.cni.namespace=kube-system --set values.cni.cniBinDir=/var/lib/cni/bin --set values.cni.cniConfDir=/etc/cni/multus/net.d --set values.cni.chained=false --set values.cni.cniConfFileName=istio-cni.conf --set values.sidecarInjectorWebhook.injectedAnnotations.k8s\.v1\.cni\.cncf\.io/networks=istio-cni   `

```
- Processing resources for Istio core.
��� Istio core installed
- Processing resources for Istiod.

- Processing resources for Istiod. Waiting for Deployment/istio-system/istiod
��� Istiod installed
- Processing resources for CNI, Egress gateways, Ingress gateways.
- Processing resources for CNI, Egress gateways, Ingress gateways. Waiting for DaemonSet/kube-system/istio-cni-node

- Processing resources for CNI, Egress gateways, Ingress gateways. Waiting for DaemonSet/kube-system/istio-cni-node, Deployment/istio-system/istio-egressgateway

- Processing resources for CNI, Egress gateways, Ingress gateways. Waiting for DaemonSet/kube-system/istio-cni-node, Deployment/istio-system/istio-egressgateway, Deployment/istio-system/istio-ingressgateway
��� Egress gateways installed
- Processing resources for CNI, Ingress gateways. Waiting for DaemonSet/kube-system/istio-cni-node, Deployment/istio-system/istio-ingressgateway
��� Ingress gateways installed

- Processing resources for CNI. Waiting for DaemonSet/kube-system/istio-cni-node
��� CNI encountered an error: failed to wait for resource: resources not ready after 5m0s: timed out waiting for the condition
DaemonSet/kube-system/istio-cni-node
- Pruning removed resourcesError: failed to install manifests: errors occurred during operation
Failed to install Istio with profile [demo]. Will retry in 10 seconds...

- Processing resources for Istio core.
��� Istio core installed
- Processing resources for Istiod.
��� Istiod installed
- Processing resources for CNI, Egress gateways, Ingress gateways.
- Processing resources for CNI, Egress gateways, Ingress gateways. Waiting for DaemonSet/kube-system/istio-cni-node
��� Egress gateways installed

��� Ingress gateways installed

- Processing resources for CNI. Waiting for DaemonSet/kube-system/istio-cni-node
��� CNI encountered an error: failed to wait for resource: resources not ready after 5m0s: timed out waiting for the condition
DaemonSet/kube-system/istio-cni-node
- Pruning removed resourcesError: failed to install manifests: errors occurred during operation
Failed to install Istio with profile [demo]. Will retry in 10 seconds...

- Processing resources for Istio core.
��� Istio core installed

- Processing resources for Istiod.
��� Istiod installed
- Processing resources for CNI, Egress gateways, Ingress gateways.
��� Egress gateways installed
- Processing resources for CNI, Ingress gateways.
- Processing resources for CNI, Ingress gateways. Waiting for DaemonSet/kube-system/istio-cni-node

��� Ingress gateways installed

- Processing resources for CNI. Waiting for DaemonSet/kube-system/istio-cni-node
��� CNI encountered an error: failed to wait for resource: resources not ready after 5m0s: timed out waiting for the condition
DaemonSet/kube-system/istio-cni-node

- Pruning removed resourcesError: failed to install manifests: errors occurred during operation
```

**Environment where the bug was observed (cloud vendor, OS, etc)**
OCP 4.7.19

istio-cni-node pod log:
[istio-cni-node-9wpc5-install-cni.log](https://github.com/istio/istio/files/6856151/istio-cni-node-9wpc5-install-cni.log)

### PR DESCRIPTION
**Pull request by jacob-delgado**
**Please provide a description of this PR:**

Fixes https://github.com/istio/istio/issues/34211 for Helm istio-cni charts, as the first only addressed istioctl.

### CODE REVIEW

**Comment by jacob-delgado** [comment]
Copying @dgn and @jwendell as this affects OpenShift when using Helm charts.

File: manifests/charts/istio-cni/templates/daemonset.yaml [hunk 1]
```diff
@@ -82,6 +82,7 @@ spec:
             runAsGroup: 0
             runAsUser: 0
             runAsNonRoot: false
+            privileged: {{ .Values.cni.privileged | quote }}
```
Comment (howardjohn) [inline manifests/charts/istio-cni/templates/daemonset.yaml:85]: is quote right? shouldn't it be `true` not `"true" `?

Comment (jacob-delgado) [reply manifests/charts/istio-cni/templates/daemonset.yaml:85]: You are right. I got this mixed up with env values.

**Comment by jacob-delgado** [comment]
cc @bianpengyuan This is the equivalent of https://github.com/istio/istio/pull/34232

File: manifests/charts/istio-cni/values.yaml [hunk 2]
```diff
@@ -34,6 +34,9 @@ cni:
   # Some k8s flavors (e.g. OpenShift) do not support the chain approach, set to false if this is the case
   chained: true
 
+  # Allow the istio-cni container to run in privileged mode, needed for some platforms (e.g. OpenShift
```
Comment (bianpengyuan) [inline manifests/charts/istio-cni/values.yaml:37]:
```suggestion
  # Allow the istio-cni container to run in privileged mode, needed for some platforms (e.g. OpenShift)
```

**Review by bianpengyuan — APPROVED** [review]

**Review by Monkeyanator — APPROVED** [review]

**Comment by jwendell** [comment]
@jacob-delgado  So you can now remove the workaround in the openshift profile with a proper value under cni:
https://github.com/istio/istio/blob/master/manifests/profiles/openshift.yaml#L13

**Pull request merged**